04.ls/ls.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+COLUMN_NUMBER = 3
+
+def main
+  directory_files = Dir.children(ARGV[0] || '.').reject { |file| file[0] == '.' }.sort
+  max_length = directory_files.map(&:length).max
+  show_ls(directory_files, max_length)
+end
+
+def show_ls(directory_files, max_length)
+  split_files(directory_files).transpose.each do |row_files|
+    row_files.each { |col_file| col_file == row_files.last ? print(col_file) : printf("%-#{max_length}s\t", col_file) }
```

Comment:
```suggestion
    row_files.each do |col_file|
      col_file == row_files.last ? print(col_file) : printf("%-#{max_length}s\t", col_file)
    end
```

横に長いのでdo/endで改行を入れた方が読みやすそうです。

あと、`col_file == row_files.last`の部分ですが、同じ文字列かどうかを比較するよりも「3番目かどうか？」を比較する方が処理効率が良さそうです。

Reply:
`each.with_index`として、文字列ではなく回数で比較するようにしました。
```ruby
row_files.each.with_index(1) do |col_file, index|
　index == row_files.length ? print(col_file) : printf("%-#{max_length}s\t", col_file)
end
```